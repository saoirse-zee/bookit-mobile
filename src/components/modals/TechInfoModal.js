import React from 'react'
import { View, StyleSheet } from 'react-native'
import { Constants } from 'expo'
import ModalContent from './ModalContent'
import { LabelText, MonoText } from '../StyledText'
import config from '../../../config.json'

const TechInfoModal = ({ onOkayPress }) => (
  <ModalContent
    title="Technical info"
    onOkayPress={onOkayPress}
  >
    <View style={styles.section}>
      <LabelText>Bookit API</LabelText>
      <MonoText>{ config.bookitApiBaseUrl }</MonoText>
    </View>
    <View style={styles.section}>
      <LabelText>Nonce</LabelText>
      <MonoText>{ config.nonce || 'Nonce has not been set.' }</MonoText>
    </View>
    <View style={styles.section}>
      <LabelText>Session ID</LabelText>
      <MonoText>{ Constants.sessionId }</MonoText>
    </View>
  </ModalContent>
)

export default TechInfoModal

const styles = StyleSheet.create({
  section: {
    marginBottom: 20,
  },
  label: {
    color: '#555555',
    width: 60,
  },
})
