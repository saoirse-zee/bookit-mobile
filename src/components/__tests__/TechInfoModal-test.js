import 'react-native'
import React from 'react'
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer'
import TechInfoModal from '../modals/TechInfoModal'

it('renders correctly', () => {
  const tree = renderer.create(<TechInfoModal
    onOkayPress={() => {}}
  />).toJSON()

  expect(tree).toMatchSnapshot()
})
