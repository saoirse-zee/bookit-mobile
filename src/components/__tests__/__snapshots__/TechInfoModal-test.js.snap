// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#4F97FF",
        "fontSize": 30,
        "fontWeight": "bold",
        "marginBottom": 50,
      }
    }
  >
    Technical info
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  />
  <View
    style={
      Object {
        "marginBottom": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          undefined,
          Object {
            "color": "#555555",
          },
        ]
      }
    >
      Bookit API
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          undefined,
          Object {
            "fontFamily": "space-mono",
          },
        ]
      }
    >
      https://staging-bookit-api.buildit.tools/v1/
    </Text>
  </View>
  <View
    style={
      Object {
        "marginBottom": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          undefined,
          Object {
            "color": "#555555",
          },
        ]
      }
    >
      Nonce
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          undefined,
          Object {
            "fontFamily": "space-mono",
          },
        ]
      }
    >
      Nonce has not been set.
    </Text>
  </View>
  <View
    style={
      Object {
        "marginBottom": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          undefined,
          Object {
            "color": "#555555",
          },
        ]
      }
    >
      Session ID
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          undefined,
          Object {
            "fontFamily": "space-mono",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": 50,
        "padding": 20,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#4F97FF",
            "fontSize": 24,
          }
        }
      >
        Okay
      </Text>
    </View>
  </View>
</View>
`;
